mozregression/prefetch.py

```diff
@@ -0,0 +1,461 @@
+        Returns build_regex and build_info_regex
+        """
+        common_config = CommonConfig(self.app_name)
+        if 'linux64' in url:
+            return (
+                [re.compile(common_config.build_regex('linux', 64))],
+                [re.compile(common_config.build_info_regex('linux', 64))])
+        elif 'linux' in url:
+            return (
+                [re.compile(common_config.build_regex('linux', 32))],
+                [re.compile(common_config.build_info_regex('linux', 32))])
+        elif 'win32' in url:
+            return (
+                [re.compile(common_config.build_regex('win', 32))],
+                [re.compile(common_config.build_info_regex('win', 32))])
+        elif 'win64' in url:
+            return (
+                [re.compile(common_config.build_regex('win', 64))],
+                [re.compile(common_config.build_info_regex('win', 64))])
+        elif 'macosx64' in url:
+            return (
+                [re.compile(common_config.build_regex('mac', 0))],
+                [re.compile(common_config.build_info_regex('mac', 0))])
+        elif self.app_name == 'fennec':
+            return (
+                [re.compile(common_config.build_regex('os', 0))],
+                [re.compile(common_config.build_info_regex('os', 0))])
+
+    def get_inbound_info(self):
+
+        inbound_info = []
+
+        if self.app_name == 'firefox':
+            app = FirefoxInboundConfigMixin()
+        elif self.app_name == 'b2g':
+            app = B2GInboundConfigMixin()
+        elif self.app_name == 'fennec':
+            app = FennecInboundConfigMixin()
+
+        os_inbound_url = app.inbound_base_urls()
+        print 'os_inbound_url: %s\n' % os_inbound_url
+        for urls in os_inbound_url:
+            builds = url_links(urls)
+            for build_url in builds:
+                build_regex, build_info_regex = self.get_regex(urls)
+                build_info_folder = BuildFolderInfoFetcher(
+                    build_regex, build_info_regex)
+                build_info = build_info_folder.find_build_info(urls+build_url)
+                print 'build_info: %s\n' % build_info
+                for builds in build_info:
+                    result = {}
+                    if 'build_txt_url' not in builds:
+                        continue
+                    build_info_text = build_info_folder.find_build_info_txt(
+                        builds['build_txt_url'])
+                    result['build_url'] = builds['build_url']
+                    result['date'] = build_info_text['date']
+                    result['changeset'] = build_info_text['changeset']
+                    result['repository'] = build_info_text['repository']
+                    inbound_info.append(result)
+
+        return inbound_info
+
+    def build_json_files(self):
+        j = json.dumps(self.get_inbound_info(), indent=2)
+        f = open('inbound_%s.json' % self.app_name, 'w')
+        print >> f, j
+        f.close
```

Comment:
This line doesn't do anything, you have to call f.close().

I'd probably actually use the with statement here for more conciseness, something like:

with open('inbound_%s.json' % self.app_name, 'w') as f:
   f.write(json.dumps(self.get_inbound_info())